### PR TITLE
Resolve several indexer target semantic errors

### PR DESCRIPTION
**Pull request by jerrymarino**
This patch resolves a few index target errors.

- external paths in Bazel are exported as "..". Rewrite these dots to external
  in all places. PCH, Framework Search Paths, Header Search Paths, etc

- patch custom compiler arguments ( copts ) for external

- redacting defines like "__TIME__" will cause a warning to be triggered, and
  they aren't useful in Xcode anyways. Remove Pinterest specific hack for this
  which just removed -Werror.

Altogether, these issues are causing many errors for indexer invocations which
are resolved as of this patch.

Change-Id: Iba5aada6203e507f4b74daaad514ae5d464ce9cd

### CODE REVIEW

**Review by bkase — APPROVED** [review]
Looks gutten!

**Review by rahul-malik — APPROVED** [review]

File: src/TulsiGenerator/PBXTargetGenerator.swift [hunk 1]
```diff
@@ -108,6 +108,14 @@ extension PBXTargetGeneratorProtocol {
   }
 }
 
+/// Assume external paths in Bazel are exported as ".."
+private func patchLeadingDotsForExternal(_ value: String) -> String {
+  if value.hasPrefix("..") {
```
Comment (jerrymarino) [inline src/TulsiGenerator/PBXTargetGenerator.swift:113]: This is no longer necessary given that the latest version of Bazel is no longer using dots, and already exports these files as `external`.

File: src/TulsiGenerator/PBXTargetGenerator.swift [hunk 2]
```diff
@@ -1036,7 +1043,8 @@ final class PBXTargetGenerator: PBXTargetGeneratorProtocol {
 
     var allOtherCFlags = data.otherCFlags
     if !data.preprocessorDefines.isEmpty {
-      allOtherCFlags.append(contentsOf: data.preprocessorDefines.sorted().map({"-D\($0)"}))
+      let filteredDefines = data.preprocessorDefines.filter { ($0.hasPrefix("__TIME") || $0.hasPrefix("__DATE")) == false }.sorted().map({"-D\($0)"})
```
Comment (jerrymarino) [inline src/TulsiGenerator/PBXTargetGenerator.swift:1046]: This is something that we may want to propose to Tulsi

File: src/TulsiGenerator/PBXTargetGenerator.swift [hunk 3]
```diff
@@ -546,21 +554,21 @@ final class PBXTargetGenerator: PBXTargetGeneratorProtocol {
...
             var path = opt.substring(from: opt.characters.index(opt.startIndex, offsetBy: 2))
             if !path.hasPrefix("/") {
+              path = patchLeadingDotsForExternal(path)
               path = "$(\(PBXTargetGenerator.BazelWorkspaceSymlinkVarName))/\(path)"
             }
             localIncludes.add(path)
           } else {
-            otherCFlags.add(opt)
+            // Patch opts like `-isystem external/` or `-iquote external`
+            let patchedOpt = opt.replacingOccurrences(of: " external/", with: " $(\(PBXTargetGenerator.BazelWorkspaceSymlinkVarName)/external/")
```
Comment (jerrymarino) [inline src/TulsiGenerator/PBXTargetGenerator.swift:570]: Also, this is something we may want to propose to tulsi

**Pull request merged**